<?xml version="1.0"?>
<launch>

	<!-- arg -->
	<arg name="gui" 		default="False" />
	<arg name="hand_frame_vision" 	value="/hand_root"/>
	<arg name="world_frame" 	value="/world"/>
        <arg name="hand_frame"		value="/hand"/>
	<arg name="record_folder"	value="$(find record)/saved/"/>
	<arg name="traj_dir"	        value="/home/guillaume/MatlabWorkSpace/peg_in_hole/TextData/"/>
	<param name="use_gui" 		value="$(arg gui)"/>

	<arg name="socket_type"		default="table_sw_one"/>


	<arg name="table_urdf" value="'$(find peg_in_hole)/urdf/One/table_sw_one.urdf'" />
	<!-- table socket -->
	<include file="$(find peg_in_hole)/launch/table/$(arg socket_type).launch">
	  	<arg name="use_rviz" 			value="False"/>
		<arg name="origin"   			value="0 0 0"/>
		<arg name="orientation"			value="0 0 0 1"/>
		<arg name="load_saved"			value="False"/>	
		<arg name="use_vision_broadcaster" 	value="False"/>
		<arg name="use_broadcaster" 		value="True"/>
	</include> 


	<!-- peg in hole-->

	<include file="$(find peg_in_hole)/launch/plug/plug_handle.launch">
	  	<arg name="use_rviz" 			value="False"/>
		<arg name="origin"   			value="0.2 0.25 0"/>
		<arg name="orientation"			value="0 0 0 1"/>
		<arg name="load_saved"			value="True"/>		
		<arg name="use_vision_broadcaster" 	value="True"/>
		<arg name="use_broadcaster" 		value="False"/>
	</include>	
 

	

	<!-- rviz -->
	<node name="rviz" pkg="rviz" type="rviz"  args="-d $(find peg_in_hole)/launch/rviz/peg_in_hole_filter_$(arg socket_type).rviz"  required="true" />

	<!-- launch optitrack -->
	<include file="$(find optitrack)/launch/optitrack.launch"/>

	<!-- launch force torque sensor listener-->
	<!--include file="$(find netft_rdt_driver)/launch/ft_sensor.launch"/-->

	<!-- replay  -->
	<!--node name="peg_in_hole_replay" pkg="peg_in_hole" type="peg_replay" args="-traj_path $(arg traj_dir) -rate 100" output="screen"/-->

	<!-- feature wrapper  --><!-- launch-prefix="xterm -e gdb -args"-->
	<node name="peg_in_hole_wrapper" pkg="peg_in_hole" type="peg_in_hole_wrapper_node" args="-urdf $(arg table_urdf) -rate 100" output="screen"/>



</launch>
